### DISCUSSION TITLE
Parallel computing

### DISCUSSION THREAD

**Discussion by POLonge** [idaholab/moose#27025] [Q&A General]
I am having this error in my simulation run:

Time Step 1, time = 1, dt = 1

    Computing Initial Residual
      Computing Residual.                                                                [ 14.24 s] [ 2261 MB]
    Finished Computing Initial Residual                                                  [ 14.24 s] [ 2261 MB]
    Computing Residual.                                                                  [ 14.33 s] [ 2261 MB]
 0 Nonlinear |R| = 1.028114e+03
    Computing Jacobian....... ..double free or corruption (out)
Aborted (core dumped)

I have a machine with 24 cores and 96GB RAM however simulation is done with only 1 core. How do I tell MOOSE to run this simulation with more cores at same time.

**Comment by GiudGiud** [comment]
Hello

You can do
```
mpirun -n 12 <your executable> -i <name of your input file>
```

**Reply by POLonge** [reply]
I tried this, it's still displaying same error.

**Reply by GiudGiud** [reply]
Ok can you compile in debug mode and run with the debug executable 

Also can you watch the memory to make sure you're running out of memory? With top on Linux

**Reply by POLonge** [reply]
I saw this message:
Time Step 1, time = 1, dt = 1

    Computing Initial Residual
      Computing Residual....................                                             [109.69 s] [14171 MB]
    Finished Computing Initial Residual                                                  [109.69 s] [14171 MB]
    Computing Residual.....................                                              [117.11 s] [14171 MB]
 0 Nonlinear |R| = 5.157454e+02
    Computing Jacobian........................................................................................free(): invalid next size (fast)
Aborted (core dumped)

===================================================================================
BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
EXIT CODE: 9
CLEANING UP REMAINING PROCESSES
YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions

Then I saw this from the documentation: A "Segmentation fault," "Segfault," or "Signal 11" error denotes a memory bug (often array access out of bounds). What's this saying?

**Reply by GiudGiud** [reply]
Looks like a genuine bug at this point.

Can you share your inputs and your mesh?
Or else could you please make a small case showing the problem

**Reply by POLonge** [reply]
Attached below is the input file.

[3D_cvf.i.zip](https://github.com/idaholab/moose/files/14547367/3D_cvf.i.zip)


Thanks.

**Reply by GiudGiud** [reply]
you are running out of memory
Code 9 for the exit code means something killed the process. This is pretty much always OOM.

Can you try with nx = ny = nz = 10 instead of 25 and see if you can get the calculation to finish

**Reply by GiudGiud** [reply]
I hit an error in devel mode
```

Assertion `i < _size' failed
Access out of bounds in MooseArray (i: 8 size: 8)
at /Users/giudgl/projects/moose/framework/build/header_symlinks/MooseArray.h, line 276
Stack frames: 14
0: 0   libmesh_devel.0.dylib               0x000000010c9ca2d0 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char>>&) + 1044
1: 1   libporous_flow-devel.0.dylib        0x0000000102fc8570 MooseArray<double>::operator[](unsigned int) const + 688
2: 2   libporous_flow-devel.0.dylib        0x0000000103072358 PorousFlowTemperatureTempl<false>::computeQpProperties() + 52
3: 3   libmoose-devel.0.dylib              0x00000001046862b0 MaterialBase::initStatefulProperties(unsigned int) + 80
4: 4   libmoose-devel.0.dylib              0x00000001046dc948 MaterialPropertyStorage::initStatefulProps(unsigned int, std::__1::vector<std::__1::shared_ptr<MaterialBase>, std::__1::allocator<std::__1::shared_ptr<MaterialBase>>> const&, unsigned int, libMesh::Elem const&, unsigned int) + 168
5: 5   libmoose-devel.0.dylib              0x0000000104b18af0 ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>>::operator()(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool) + 324
6: 6   libmoose-devel.0.dylib              0x0000000104ccc674 void libMesh::Threads::parallel_reduce<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, ComputeMaterialsObjectThread>(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, ComputeMaterialsObjectThread&) + 116
7: 7   libmoose-devel.0.dylib              0x0000000104ccc55c FEProblemBase::initElementStatefulProps(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool) + 84
8: 8   libmoose-devel.0.dylib              0x0000000104d032d4 FEProblemBase::initialSetup() + 15164
9: 9   libmoose-devel.0.dylib              0x000000010461dde0 Transient::init() + 88
10: 10  libmoose-devel.0.dylib              0x0000000105671f90 MooseApp::executeExecutioner() + 660
11: 11  libmoose-devel.0.dylib              0x000000010566a58c MooseApp::run() + 4584
12: 12  relap-7-devel                       0x0000000100ca7b64 main + 156
13: 13  dyld                                0x000000018a4d50e0 start + 2360
[0] /Users/giudgl/projects/moose/framework/build/header_symlinks/MooseArray.h, line 276, compiled Mar  8 2024 at 16:16:16
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

**Reply by GiudGiud** [reply]
So it might not be oom. I ll take a look probably tomorrow

**Reply by GiudGiud** [reply]
ok I looked into this a little. 
Porous flow materials aren't coded for second order variables. 
You should make elem_type = HEX8 (or comment it out) then change the displacements to be first order.

If you want to use second order variables, you'll need to adapt quite a bit of logic in the porous flow materials, which I think is there for nodal lumping
